app/View/Components/Forms/EnvVarInput.php

```
@@ -0,0 +1,73 @@
+        public bool $autoDisable = true,
+        public array $availableVars = [],
+    ) {
+        // Handle authorization-based disabling
+        if ($this->canGate && $this->canResource && $this->autoDisable) {
+            $hasPermission = Gate::allows($this->canGate, $this->canResource);
+
+            if (! $hasPermission) {
+                $this->disabled = true;
+            }
+        }
+    }
```

PHPMD (2.15.0)

The method __construct has 18 parameters. Consider reducing the number of parameters to less than 10. (undefined)

(ExcessiveParameterList)